troubleshooting/tools/tracevol.py

```diff
@@ -362,28 +470,43 @@ def get_pool_name(arg, vol_id):
     stdout, stderr = out.communicate()
     if stderr is not None:
         if arg.debug:
-            print("failed to pool name %s", stderr)
+            print("failed to get the pool name %s", stderr)
         return ""
     try:
         pools = json.loads(stdout)
     except ValueError as err:
         if arg.debug:
-            print("failed to pool name %s", err)
+            print("failed to get the pool name %s", err)
         return ""
-    pool_id = vol_id.split('-')
-    if len(pool_id) < 4:
-        if arg.debug:
-            print("pood id notin proper format", pool_id)
-        return ""
-    if pool_id[3] in arg.rooknamespace:
-        pool_id = pool_id[4]
+    if isRBD:
+        pool_id = vol_id.split('-')
+        if len(pool_id) < 4:
+            if arg.debug:
+                print("pood id not in the proper format", pool_id)
+                return ""
+        if pool_id[3] in arg.rooknamespace:
+            pool_id = pool_id[4]
+        else:
+            pool_id = pool_id[3]
+        for pool in pools:
+            if int(pool_id) is int(pool['poolnum']):
+                return pool['poolname']
     else:
-        pool_id = pool_id[3]
-    for pool in pools:
-        if int(pool_id) is int(pool['poolnum']):
-            return pool['poolname']
+        for pool in pools:
+            return pool['metadata_pool']
     return ""
 
+'''
+myDict contains key value store for annotations.
+one of the annotations value is provisioner name.
+provisioner name should tell whether pvc is for
+cephfs or rbd.
+'''
+def isRBDPV(myDict, item):
+    for key in myDict:
```

Comment:
This could be expressed more tersely with use of the any() function, but this is no big deal.